File: VB/DXWebApplication1/Controllers/HomeController.vb
Imports System
Imports System.ComponentModel.DataAnnotations
Imports System.Web.Mvc
Imports DevExpress.Web

Namespace DXWebApplication1.Controllers
	Public Class UserModelDTO
		<Required>
		Public Property NameProperty() As String
		Public Property ImagesProperty() As UploadedFile()
	End Class

    Public Class UserModel
        Public Property Name() As String
        Public Property Avatar() As String
    End Class

	Public Class HomeController
		Inherits Controller

		<HttpGet>
		Public Function Index() As ActionResult
			Return View()
		End Function

		<HttpPost>
		Public Function Index(ByVal modelDTO As UserModelDTO) As ActionResult
			Dim avatarFileName As String = String.Empty
			If ModelState.IsValid Then
				If modelDTO.ImagesProperty.Length > 0 AndAlso modelDTO.ImagesProperty(0).ContentLength > 0 Then
					avatarFileName = String.Format("~/Content/Images/{0}", modelDTO.ImagesProperty(0).FileName)
					modelDTO.ImagesProperty(0).SaveAs(Server.MapPath(avatarFileName))
				End If
			End If

			Dim model As New UserModel()
			model.Name = modelDTO.NameProperty
			model.Avatar = avatarFileName

			Return View("Complete", model)
		End Function
	End Class
End Namespace

File: CS/DXWebApplication1/Controllers/HomeController.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.Web.Mvc;
using DevExpress.Web;

namespace DXWebApplication1.Controllers {
    public class UserModelDTO {
        [Required]
        public string NameProperty { get; set; }
        public UploadedFile[] ImagesProperty { get; set; }
    }

    public class UserModel {
        public string Name { get; set; }
        public string Avatar { get; set; }
    }

    public class HomeController: Controller {
        [HttpGet]
        public ActionResult Index() {
            return View();
        }

        [HttpPost]
        public ActionResult Index(UserModelDTO modelDTO) {
            string avatarFileName = string.Empty;
            if (ModelState.IsValid) {
                if (modelDTO.ImagesProperty.Length > 0 && modelDTO.ImagesProperty[0].ContentLength > 0) {
                    avatarFileName = string.Format("~/Content/Images/{0}", modelDTO.ImagesProperty[0].FileName);
                    modelDTO.ImagesProperty[0].SaveAs(Server.MapPath(avatarFileName));
                }
            }

            UserModel model = new UserModel();
            model.Name = modelDTO.NameProperty;
            model.Avatar = avatarFileName;

            return View("Complete", model);
        }
    }
}